### DISCUSSION TITLE
how to set default value in q-popup-edit

### DISCUSSION THREAD

**Discussion by chiran22** [quasarframework/quasar#14390] [General - Components / Directives / etc] [answered]
Hi,
   I am trying to set a default value in q-popup-edit. I see the setting of :model-value property but I can not set this by clicking enter. Here is my code. Please help me.

<q-popup-edit v-else-if="getIfColumnIsEditable(prop.name)"
                    v-model="props.row[prop.name]"
                    auto-save
                    v-slot="scope"
    
    
                  >
                    <q-input
                      v-model="scope.value"
                      step=10
                      type="number"
                      dense
                      min=10
                      @keyup.enter="scope.set"
                      autofocus
                      :label="getHeader(prop.name, 1)"

                         />
                  </q-popup-edit>

**Comment by MilosPaunovic** [comment, marked as answer]
Take a look at the example on the docs:
https://quasar.dev/vue-components/table#popup-editing

**Reply by chiran22** [reply]
Thank you for the response. I checked the examples but I don't see setting up default value. I want to see the default value like "10" when the user clicks on the control. the data coming from database and if that field is null and when the user clicks, user should see the value 10 by default and user should be able to set the value.

**Reply by MilosPaunovic** [reply]
Try with something like this:
https://codepen.io/milospaunovic/pen/gOzmdGB?editors=1010

**Reply by chiran22** [reply]
Thank you very much. Its showing the default value but When I enter or click outside of the control, The default value is not getting set. I need to click up/down arrow to set the value. Here is my code. The default value should be set when I click enter or click outside control.

<q-input
                      v-model="scope.value"
                      step=10
                      type="number"
                      dense
                      min=10
                      @keyup.enter="scope.set"
                      autofocus
                      :label="getHeader(prop.name, 2)"
                      :model-value = "scope.value ? scope.value : 10"
                    />

**Reply by MilosPaunovic** [reply]
React to `PopupEdit` events, such as`hide` or `cancel` and call your sav method there.

**Reply by chiran22** [reply]
Thanks a lot for your responses. Could you please please write some code and send me. I am very very new to this.

**Reply by MilosPaunovic** [reply]
If you are new to this, you need to write some code yourself, that is the best path to understanding. Use Codepen examples from AUasar documentation as a starting point.

**Reply by chiran22** [reply]
Hi,
    I tried the examples but I am not able to achieve what I want. Could you please please send me some samples on this events how to raise them from control?
